feat(chore): implement revoke token and revoke all token

jorisdugue opened the pull request:
- [X] Refactoring of some imports.
- [X] Implementation of the revoke Token to support the following RFCs for the different oauth2 (website) allowing it :
- https://oauth.net/2/token-revocation/ (from https://github.com/lelylan/simple-oauth2/blob/9dbe7d26353793c8cbdfb3a83d754120ef36ccc1/lib/access-token.js#L61C12-L61C12)
- https://www.rfc-editor.org/rfc/rfc7009.html
- [X] Implement revokeAllToken from  https://github.com/lelylan/simple-oauth2/blob/9dbe7d26353793c8cbdfb3a83d754120ef36ccc1/lib/access-token.js#L81

This branch has direct dependencies with #220 

close #221 
<!--
Thank you for your pull request. Please provide a description above and review
the requirements below.

Bug fixes and new features should include tests and possibly benchmarks.

Contributors guide: https://github.com/fastify/fastify/blob/master/CONTRIBUTING.md

By making a contribution to this project, I certify that:

* (a) The contribution was created in whole or in part by me and I
  have the right to submit it under the open source license
  indicated in the file; or

* (b) The contribution is based upon previous work that, to the best
  of my knowledge, is covered under an appropriate open source
  license and I have the right under that license to submit that
  work with modifications, whether created in whole or in part
  by me, under the same open source license (unless I am
  permitted to submit under a different license), as indicated
  in the file; or

* (c) The contribution was provided directly to me by some other
  person who certified (a), (b) or (c) and I have not modified
  it.

* (d) I understand and agree that this project and the contribution
  are public and that a record of the contribution (including all
  personal information I submit with it, including my sign-off) is
  maintained indefinitely and may be redistributed consistent with
  this project or the open source license(s) involved.
-->

#### Checklist

- [x] run `npm run test` and `npm run benchmark`
- [x] tests and/or benchmarks are included
- [x] documentation is changed or added
- [x] commit message and code follows the [Developer's Certification of Origin](https://github.com/fastify/.github/blob/master/CONTRIBUTING.md#developers-certificate-of-origin-11)
      and the [Code of conduct](https://github.com/fastify/.github/blob/master/CODE_OF_CONDUCT.md)

Uzlopak commented on README.md:
````diff
@@ -254,6 +255,8 @@ fastify.get('/external', { /* Hooks can be used here */ }, async (req, reply) =>
 });
 ```
 
+- `revokeToken(Token, tokenType, params, callback)`: A function that use for remove the current access_token or refresh_token. If the callback is not passed it will return a promise. The callback call or promise resolution returns an `void`
````
```suggestion
- `revokeToken(Token, tokenType, params, callback)`: A function to revoke the current access_token or refresh_token on server side. If the callback is not passed it will return a promise. The callback call or promise resolution returns `void`
```

mcollina reviewed (changes requested):
Can you add the test for the types? We use tsd.

jorisdugue commented:
I implemented only `revoke` method from simple-oauth2

Should we also implement the `revokeAll` method which removes the two types of token (refresh_token and access_token )?
It should also be noted that not all providers necessarily take the revoke method, for example github does not take it into account

I just added the tests for the missing interfaces @mcollina

Uzlopak commented:
Yes, please add revokeAll.

mcollina reviewed (approved):
lgtm

Uzlopak reviewed (approved):
LGTM
